### DISCUSSION TITLE
How to set traceId in the root Span

### DISCUSSION THREAD

**Discussion by ajprakas** [open-telemetry/opentelemetry-java#2598] [Q&A]
I have use case to set TraceId in root span as one business objectId. Is it possible to do so ?

**Comment by anuraaga** [comment]
Hi @ajprakas - trace ID is for tracing so we don't really allow setting it manually. Generally users use `Baggage` to propogate domain-specific values. Is that something you can use?

**Reply by jkwatson** [reply]
there are several ways you could hack this, as long as your business object id is also a valid span id. 
1) Create a custom IdGenerator that does what you need
2) Create a SpanContext instance with your trace id, put it into the Context and use it as the root span's parent.

However, this seems like an odd request. Trace IDs are not supposed to be business object ids. They're supposed to be globally unique and never re-used across traces. It seems like if you're using business object ids for that, you're almost certainly going to end up with re-used traces, and that is going to be a very bad problem for almost every back end tracing system.
